### DISCUSSION TITLE
Customised order of navigation groups and navigation items

### DISCUSSION THREAD

**Discussion by retifex** [filamentphp/filament#9671] [Help]
The documentation [explains](https://filamentphp.com/docs/3.x/panels/navigation#grouping-navigation-items):

> Ungrouped items will remain at the top of the sidebar.

I use top navigation, it is the same there; in this case the ungrouped items remain on the left side.

How can I customise the order of navigation groups and navigation items in the navigation?

For example, I would like this order:

[Navigation Group]
[Custom Navigation Item]
[Navigation Group]

I tried to give my navigation item a high sort value, but to no avail:

```php
->navigationItems([
    NavigationItem::make('Github')
        ->url('https://github.com', shouldOpenInNewTab: true)
        ->sort(99),
])
```

And obviously, navigation groups do not have a $navigationSort property.

How do you achieve the above sequence?

**Comment by leandrocfe** [comment]
[this](https://filamentphp.com/docs/3.x/panels/navigation#ordering-navigation-groups)?

**Reply by retifex** [reply]
Your link points to an explanation for ordering groups. Like I said, the custom navigation item is not a group, but a single link.

Trying to order the single link the same way as ordering groups has no effect:

```php
->navigationItems([
    NavigationItem::make('Custom-Group-1')
        ->url('https://microsoft.com', shouldOpenInNewTab: true)
        ->icon('heroicon-o-rectangle-stack')
        ->group('Custom-Group-1')
        ->sort(1),
    NavigationItem::make('Custom-Group-3')
        ->url('https://microsoft.com', shouldOpenInNewTab: true)
        ->icon('heroicon-o-rectangle-stack')
        ->group('Custom-Group-3')
        ->sort(3),
    NavigationItem::make('Custom-Group-5')
        ->url('https://microsoft.com', shouldOpenInNewTab: true)
        ->icon('heroicon-o-rectangle-stack')
        ->group('Custom-Group-5')
        ->sort(5),
])
->navigationItems([
    NavigationItem::make('Custom-Link')
        ->url('https://microsoft.com', shouldOpenInNewTab: true)
        ->icon('heroicon-o-rectangle-stack')
        ->sort(2),
])
->navigationGroups([
    'Custom-Group-1',
    'Custom-Link',
    'Custom-Group-3',
    'Resource-Group',
    'Custom-Group-5',
])
```

![image](https://github.com/filamentphp/filament/assets/72259022/cad4dc29-97b0-4883-bb5c-ea6b96ea04aa)

The only workaround I can think of is to turn my single link into a group, but that's not the solution I'm looking for.

**Reply by leandrocfe** [reply]
Hum, I've never tried it before...

**Comment by PaulCoullAaglobal** [comment]
@retifex Did you manage to solve the issue without adding them to a group?

I know it seems complicated but would it be possible to create several navigation on the left and order it that way.

**Comment by 3adeling** [comment]
You can make group as item by setting the label of the group to null

```
NavigationGroup::make('Home')
    ->label(null)
```

**Comment by commey** [comment]
Any update in this? 
Navigation groups seem to be always positioned at the end after navigation items have been placed.
I need to be able to set the order for a navigation group and not the items in the navigation group.

**Comment by onemoreahmad** [comment]
Check https://github.com/filamentphp/filament/discussions/11570#discussioncomment-15472729
